packages/vm/src/parsed_wasm.rs

```diff
@@ -61,10 +71,12 @@ impl<'a> ParsedWasm<'a> {
             match p {
                 Payload::TypeSection(t) => {
                     this.type_count = t.get_count();
-                    for t_res in t {
+                    for (type_index, t_res) in t.into_iter().enumerate() {
                         let ty: Type = t_res?;
                         match ty {
                             Type::Func(ft) => {
+                                this.type_params.insert(type_index, ft.params().len());
```

Comment:
Given that it is impossible to skip an index here, we can store this in a vector as well, right?